pkgs/development/libraries/qt-6/modules/qtwebengine.nix

```diff
@@ -0,0 +1,335 @@
+{ qtModule
+, qtdeclarative
+, qtwebchannel
+, qtpositioning
+, qtwebsockets
+, bison
+, coreutils
+, flex
+, git
+, gperf
+, ninja
+  # ninja is hard requirement for buildPhase
+  # https://bugreports.qt.io/browse/QTBUG-96897
+, pkg-config
+, python3
+, which
+, nodejs
+, qtbase
+, perl
+, xorg
+, libXcursor
+, libXScrnSaver
+, libXrandr
+, libXtst
+, libxshmfence
+, libXi
+, fontconfig
+, freetype
+, harfbuzz
+, icu
+, dbus
+, libdrm
+, zlib
+, minizip
+, libjpeg
+, libpng
+, libtiff
+, libwebp
+, libopus
+, jsoncpp
+, protobuf
+, libvpx
+, srtp
+, snappy
+, nss
+, libevent
+, openssl
+, alsa-lib
+, pulseaudio
+, libcap
+, pciutils
+, systemd
+, pipewire
+, enableProprietaryCodecs ? true
+, gn
+  # TODO(milahu) remove gn?
+  # -- Could NOT find Gn: Found unsuitable version "1718 (fd3d768)", but required is exact version "6.2.2"
+, cctools
+, libobjc
+, libunwind
+, sandbox
+, xnu
+, ApplicationServices
+, AVFoundation
+, Foundation
+, ForceFeedback
+, GameController
+, AppKit
+, ImageCaptureCore
+, CoreBluetooth
+, IOBluetooth
+, CoreWLAN
+, Quartz
+, Cocoa
+, LocalAuthentication
+, cups
+, openbsm
+, runCommand
+, xcbuild
+, writeScriptBin
+, ffmpeg
+, lib
+, stdenv
+, fetchpatch
+, glib
+, libxml2
+, libxslt
+, lcms2
+, re2
+, kerberos ? null
+}:
+
+# TODO: fix build
+# TODO(milahu) add optional dependencies? Qt6QmlCompilerPlus Qt6Designer
+
+qtModule rec {
+  pname = "qtwebengine";
+  qtInputs = [ qtdeclarative qtwebchannel qtwebsockets qtpositioning ];
+  nativeBuildInputs = [
+    bison
+    coreutils
+    flex
+    git
+    gperf
+    ninja
+    pkg-config
+    (python3.withPackages (ps: with ps; [ html5lib ]))
+    which
+    gn
+    nodejs
+  ] ++ lib.optional stdenv.isDarwin xcbuild;
+  doCheck = true;
+  outputs = [ "out" "dev" ];
+
+  # Don’t use the gn setup hook
+  dontUseGnConfigure = true;
+
+  # ninja builds some components with -Wno-format,
+  # which cannot be set at the same time as -Wformat-security
+  hardeningDisable = [ "format" ];
+  NIX_CFLAGS_COMPILE = "-fpermissive";
+
+  patches = [(fetchpatch {
+    url = "https://code.qt.io/cgit/qt/qtwebengine-chromium.git/patch/?id=75f0f4eb";
+    stripLen = 1;
+    extraPrefix = "src/3rdparty/";
+    sha256 = "sha256-3aMcVXJg+v+UbsSO27g6MA6/uVkWUxyQsMD1EzlzXDs=";
+  })];
+
+  postPatch = ''
+    # Patch Chromium build tools
+    (
+      cd src/3rdparty/chromium;
+
+      # Manually fix unsupported shebangs
+      substituteInPlace third_party/harfbuzz-ng/src/src/update-unicode-tables.make \
+        --replace "/usr/bin/env -S make -f" "/usr/bin/make -f" || true
+      substituteInPlace third_party/webgpu-cts/src/tools/deno \
+        --replace "/usr/bin/env -S deno" "/usr/bin/deno" || true
+      patchShebangs .
```

Comment:
Can we be a bit more specific which files need patching? Checking all of the many thousand files can take a bit.

Reply:
workaround for #77539